x-pack/plugins/osquery/server/routes/action/create_action_route.ts

```diff
@@ -55,8 +78,6 @@ export const createActionRoute = (router: IRouter, osqueryContext: OsqueryAppCon
         return response.badRequest({ body: new Error('No agents found for selection') });
       }
 
-      // TODO: Add check for `runSavedQueries` only
```

Comment:
this check was about a case when an user has only `runSavedQueries` permission, so then we should find an SavedQuery via `request.body.saved_query_id` and use it instead of `request.body.query` and `request.body.ecs_mapping`